submissions/Lu-sure/post-oop/index.js

```diff
@@ -0,0 +1,162 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+  // ======== OBJECTS DEFINITIONS ========
+  class Inhabitant {
+    constructor(name, age, spesies) {
+      this.name = name;
+      this.age = age;
+      this.species = spesies;
+    }
+
+    getName() {
+      return this.name;
+    }
+
+    introduce() {
+      return `I'm ${this.age} years old ${this.species}.`
+    }
+  }
+
+  class Human extends Inhabitant {
+    constructor(name, lastName, age, gender, saying) {
+      super(name, age, 'human');
+      this.gender = gender;
+      this.lastName = lastName;
+      this.saying = saying;
+    }
+
+    getName() {
+      return `${this.name} ${this.lastName}`;
+    }
+
+    introduce() {
+      return `${super.introduce()} I feel I'm ${this.gender}. My message is "${this.saying}" `;
+    }
+  }
+
+  class Woman extends Human {
+    constructor(name, lastName, age, saying) {
+      super(name, lastName, age, 'woman', saying);
+      this.lovedBy;
+    }
+
+    getOneLovedBy(lovedBy) {
+      this.lovedBy = lovedBy;
+    }
+
+    beLoved() {
+      return `I'm happy to be loved by ${this.lovedBy === this.getName() ? 'myself' : this.lovedBy}.`;
+    }
+
+    introduce() {
+      return super.introduce()+this.beLoved();
+    }
+  }
+
+  class Man extends Human {
+    constructor(name, lastName, age, saying) {
+      super(name, lastName, age, 'man', saying);
+    }
+
+    beGentleman() {
+      return `Anyway I believe, every woman is beautiful!`;
+    }
+
+    introduce() {
+      return super.introduce() + this.beGentleman();
+    }
+  }
+
+  class Pet extends Inhabitant {
+    constructor(name, age, spesies, sex) {
+      super(name, age, spesies);
+      this.sex = sex;
+    }
+
+    beTooMuchNice() {
+      return ` ..sleeping softly on the knees. `;
+    }
+
+    introduce() {
+      return super.introduce() + this.beTooMuchNice();
+    }
+  }
+
+  class Cat extends Pet {
+    constructor(name, age, sex) {
+      super(name, age, 'cat', sex);
+    }
+
+    sayMeow() {
+      return `meow!..`
+    }
+
+    introduce() {
+      return super.introduce() + this.sayMeow();
+    }
+  }
+
+  class Dog extends Pet {
+    constructor(name, age, sex) {
+      super(name, age, 'dog', sex);
+    }
+
+    defendMyLord() {
+      return `Go away, strangers! woof-woof..`
+    }
+
+    introduce() {
+      return super.introduce() + this.defendMyLord();
+    }
+  }
+
+  function add_prop_value(o, S, prop) { o[prop] += ` & ${S[prop]}` };
```

Comment:
What are `o` and `S` in fact? Quite mysterious naming

Reply:
Got my mistake, made changes.